### PR TITLE
doc, java, update onboard doc

### LINKED ISSUE #2657: [Document]: Improve the onboarding doc for DPG sdk-automation.
The spec is out of date which only talked about autorest configuration.
Please update it for typespec scenarios.

https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/onboard-dpg-in-sdkautomation/java/README.md

Follow up on issue: https://github.com/Azure/azure-sdk-tools/issues/7818

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
fix https://github.com/Azure/autorest.java/issues/2657

relate to https://github.com/Azure/azure-sdk-tools/issues/7818

# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: documentation/onboard-dpg-in-sdkautomation/java/README.md [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
+    service-name: "Standby Pool"
+    examples-directory: "examples"
 ```
-~~~
-- `namespace`: The namespace of generated sdk.
-- `output-folder`: The relative path of destination to generate SDK.
-- `require`: The item of the value is the relative path of spec readme.md file.
-- `package-A` and `package-B` must be defined in swagger `readme.md` file. For samples, please refer to [dataplane samples for multi client](../../samplefiles-dp/samplefiles-dp-for-multi-client).
+
+# Add AutoRest Configuration of Java SDK
```
Comment (XiaofeiCao) [inline documentation/onboard-dpg-in-sdkautomation/java/README.md:39]:
```suggestion
# Add AutoRest Configuration for Java SDK 
```

File: documentation/onboard-dpg-in-sdkautomation/java/README.md [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
+# Add TypeSpec Configuration for Java SDK
 
-## Parameter Description
+See [Use SDK Automation from REST API specifications](https://github.com/Azure/azure-sdk-for-java/wiki/TypeSpec-Java-Quickstart#use-sdk-automation-from-rest-api-specifications) in TypeSpec Java QuickStart.
 
-- `<ServiceName>`: The RP name, which is usually same as rp name in swagger.
-- `<PackageName>`: Java package name.
-- `<Namespace>`: It should start with `com.`, and the remaining part can be got from `<PackageName>` by replacing `- with `.'. For example, if `<PackageName>` is `azure-analytics-purview-administration`, then `<Namespace>` is `com.azure.analytics.purview.administration`.
+`flavor` is always `azure` for Azure SDK.
+`examples-directory` is same as that in e.g. typespec-autorest emitter.
```
Comment (XiaofeiCao) [inline documentation/onboard-dpg-in-sdkautomation/java/README.md:6]: Do we need to mention `service-dir`? Though it's cross language..

**Review by XiaofeiCao — APPROVED** [review]

File: documentation/onboard-dpg-in-sdkautomation/java/README.md (hunk 2 above)
Comment (weidongxu-microsoft) [reply documentation/onboard-dpg-in-sdkautomation/java/README.md:6]: Yeah, that's the reason I didn't mention this. I assume dev should get it from https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md#service-folder-structure
Emm, however that doc not explicitly talk about `service-dir`.

Anyway, we can add this later, if we see more misconfigure on that.

Comment (weidongxu-microsoft) [reply documentation/onboard-dpg-in-sdkautomation/java/README.md:6]: We've mentioned it in our QuickStart https://github.com/Azure/azure-sdk-for-java/wiki/TypeSpec-Java-Quickstart#configuration

**Pull request merged**